### PR TITLE
Add ability to listen on TCP

### PR DESCRIPTION
**Pull request by loverend**
Currently we only accept incomming messages on UDP, this adds tcp. It is
selected using `--protocol tcp`

### CODE REVIEW

**Comment by loverend** [comment]
I have updated `napalm-logs-profile` and ran against both udp and tcp for comparison:

UDP:
```
Sent 667455 messages in 60 seconds
Sent 670712 messages in 60 seconds
Sent 666473 messages in 60 seconds
Sent 670367 messages in 60 seconds
Sent 664520 messages in 60 seconds
Received 17152 messages in 50 seconds
```

TCP
```
Received 21624 messages in 50 seconds
Sent 34372 messages in 60 seconds
Sent 34660 messages in 60 seconds
Sent 34414 messages in 60 seconds
Sent 34526 messages in 60 seconds
Sent 34662 messages in 60 seconds
```

File: napalm_logs/listener.py [hunk 1]
```diff
@@ -40,23 +46,80 @@ def _setup_ipc(self):
...
+            skt.bind((self.address, self.port))
+        except socket.error as msg:
+            error_string = 'Unable to bind to port {} on {}: {}'.format(self.port, self.address, msg)
+            log.error(error_string, exc_info=True)
+            raise BindException(error_string)
+
+        while self.__up:
+	    skt.listen(1)
+	    conn, addr = skt.accept()
+            thread = threading.Thread(target=self._tcp_connection, args=(conn, addr,))
```
Comment (mirceaulinic) [inline napalm_logs/listener.py]: Does this work OK? Looks like wrong indentation.

File: napalm_logs/listener.py [hunk 2]
```diff
@@ -40,23 +46,80 @@ def _setup_ipc(self):
...
+
+    def _tcp_connection(self, conn, addr):
+        log.debug('Established connection with {}'.format(addr))
+        conn.settimeout(TIMEOUT)
+        while True:
+            try:
+                msg = conn.recv(BUFFER_SIZE)
+            except socket.timeout:
+                log.debug('Connection {} timed out'.format(addr))
+                break
```
Comment (mirceaulinic) [inline napalm_logs/listener.py]: Please avoid break. Just return or exit from the thread

File: napalm_logs/listener.py [hunk 3]
```diff
@@ -40,23 +46,80 @@ def _setup_ipc(self):
...
+            if not msg:
```
Comment (mirceaulinic) [inline napalm_logs/listener.py]: If the message is empty, again  avoid break & log this at info level

File: napalm_logs/listener.py (hunk 1 above)
Comment (loverend) [inline napalm_logs/listener.py]: It did work, but somehow I managed to add tabs rather than spaces.

**Comment by loverend** [comment]
Addressed all comments and updated.

**Pull request merged**